lib/UserResolver.php

```diff
@@ -24,15 +24,25 @@ public function __construct(IUserManager $userManager) {
 	 * @throws NoUserFoundException
 	 */
 	public function findExistingUserId(string $rawUidCandidate, bool $force = false): string {
-		if ($force) {
-			$this->ensureUser($rawUidCandidate);
+		if ($force)
+		{
+			$usersFound=$this->ensureUser($rawUidCandidate);
+			if(count($usersFound) != 1)
+			{
+				throw new NoUserFoundException('User' . $rawUidCandidate . ' not valid or not found');
+			}
+			$user = array_keys($usersFound)[0];
```

Comment:
There is a risk that more records are returned and that the first one is not the correct one.

It would be acceptable to cycle through the result and compare with the uid candidate (after sanitizing), for instance.